src/Models/ModelInstantiator.php

```diff
@@ -325,4 +330,39 @@ public function convertPropertyOnRetrieve(
 
         return $column->getConverter()->onRetrieve($property);
     }
+
+    /**
+     * Public for testability.
+     * @param string $property ReflectionProperties->getName() always returns
+     *                         a string which is the key of the array that
+     *                         handles this property.
+     * @param array $property_reflection
+     * @return \ReflectionProperty
+     * @throws \TypeError
+     */
+    public function getReflectionProperty(string $property, array $property_reflection): \ReflectionProperty
+    {
+        $reflection_index = array_search($property, array_column($property_reflection, 'name'), true);
+
+        if (false === $reflection_index || !isset($property_reflection[$reflection_index])) {
+            throw new \TypeError();
```

Comment:
Is `TypeError` the appropriate Exception type here? Maybe `OutOfBoundsException` with an appropriate message is more correct.

https://www.php.net/manual/en/class.outofboundsexception.php

Also I think we should add the use statement and get rid of the `\`